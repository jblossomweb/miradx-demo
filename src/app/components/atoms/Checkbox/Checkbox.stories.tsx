import React from 'react'
import type { Meta, StoryObj } from '@storybook/react'

import Checkbox from './Checkbox'

type Story = StoryObj<typeof meta>

const meta = {
  title: 'components/atoms/Checkbox',
  component: Checkbox,
  render: (args) => <Checkbox {...args} />,
} satisfies Meta<typeof Checkbox>

export const Sample: Story = {
  args: {
    label: 'Sample Checkbox',
  },
}

export const Disabled: Story = {
  args: {
    label: 'Disabled Checkbox',
    disabled: true,
  },
}

export const Checked: Story = {
  args: {
    label: 'Checked Checkbox',
    checked: true,
    onChange: (event) => {
      console.log(`set checked to ${event.target.checked}`)
    },
  },
}

export const Unchecked: Story = {
  args: {
    label: 'Unchecked Checkbox',
    checked: false,
    onChange: (event) => {
      console.log(`set checked to ${event.target.checked}`)
    },
  },
}

export default meta
