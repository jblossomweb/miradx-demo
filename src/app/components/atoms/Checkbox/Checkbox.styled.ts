import tw from 'tailwind-styled-components'

export interface ContainerProps {
  $disabled?: boolean
}

export const Container = tw.label<ContainerProps>`
  flex
  items-center
  font-medium
  ${({ $disabled }) => ($disabled ? 'text-gray-400' : '')}
  ${({ $disabled }) => ($disabled ? 'cursor-not-allowed' : 'cursor-pointer')}
`

export const Input = tw.input`
  w-4 h-4
  mr-2
  ${({ disabled }) => (disabled ? 'cursor-not-allowed' : 'cursor-pointer')}
  rounded
  focus:ring-blue-500
  focus:ring-1
`
