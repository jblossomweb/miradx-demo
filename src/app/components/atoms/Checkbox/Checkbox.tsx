import React from 'react'
import * as Styled from './Checkbox.styled'

export interface CheckboxProps {
  label?: string
  disabled?: boolean
  checked?: boolean
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
}

const Checkbox: React.FC<CheckboxProps> = ({
  label,
  disabled,
  checked,
  onChange,
}) => (
  <Styled.Container $disabled={disabled}>
    <Styled.Input
      type="checkbox"
      disabled={disabled}
      checked={checked}
      onChange={onChange}
    />{' '}
    {label}
  </Styled.Container>
)

export default Checkbox
