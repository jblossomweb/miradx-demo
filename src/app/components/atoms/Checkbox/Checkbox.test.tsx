import React from 'react'
import { render, screen } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import Checkbox from './Checkbox'
import * as stories from './Checkbox.stories'

describe('app/components/atoms/Checkbox', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('assertions', () => {
    const onChange = jest.fn()
    const renderDefault = () => render(<Checkbox />)
    const renderChecked = () =>
      render(
        <Checkbox
          checked={true}
          onChange={onChange}
        />,
      )
    const renderUnChecked = () =>
      render(
        <Checkbox
          checked={false}
          onChange={onChange}
        />,
      )

    beforeEach(jest.clearAllMocks)

    it('renders a checkbox', () => {
      renderDefault()
      const checkbox = screen.getByRole('checkbox')
      expect(checkbox).toBeInTheDocument()
    })

    it('renders a checkbox unchecked by default', () => {
      renderDefault()
      const checkbox = screen.getByRole('checkbox')
      expect(checkbox).toBeInTheDocument()
      expect(checkbox).not.toBeChecked()
    })

    it('renders a checked checkbox when specified', () => {
      renderChecked()
      const checkbox = screen.getByRole('checkbox')
      expect(checkbox).toBeInTheDocument()
      expect(checkbox).toBeChecked()
    })

    it('renders an unchecked checkbox when specified', () => {
      renderUnChecked()
      const checkbox = screen.getByRole('checkbox')
      expect(checkbox).toBeInTheDocument()
      expect(checkbox).not.toBeChecked()
    })

    it('calls onChange when checkbox is clicked', () => {
      renderUnChecked()
      const checkbox = screen.getByRole('checkbox')
      expect(checkbox).toBeInTheDocument()
      expect(checkbox).not.toBeChecked()
      expect(onChange).not.toHaveBeenCalled()
      checkbox.click()
      expect(onChange).toHaveBeenCalled()
    })
  })
})
