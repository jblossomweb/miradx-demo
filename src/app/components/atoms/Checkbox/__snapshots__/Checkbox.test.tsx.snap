// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Checkbox stories Checked matches markup snapshot 1`] = `
<DocumentFragment>
  <label
    class="flex items-center font-medium cursor-pointer"
  >
    <input
      checked=""
      class="w-4 h-4 mr-2 cursor-pointer rounded focus:ring-blue-500 focus:ring-1"
      type="checkbox"
    />
     Checked Checkbox
  </label>
</DocumentFragment>
`;

exports[`app/components/atoms/Checkbox stories Disabled matches markup snapshot 1`] = `
<DocumentFragment>
  <label
    class="flex items-center font-medium text-gray-400 cursor-not-allowed"
  >
    <input
      class="w-4 h-4 mr-2 cursor-not-allowed rounded focus:ring-blue-500 focus:ring-1"
      disabled=""
      type="checkbox"
    />
     Disabled Checkbox
  </label>
</DocumentFragment>
`;

exports[`app/components/atoms/Checkbox stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <label
    class="flex items-center font-medium cursor-pointer"
  >
    <input
      class="w-4 h-4 mr-2 cursor-pointer rounded focus:ring-blue-500 focus:ring-1"
      type="checkbox"
    />
     Sample Checkbox
  </label>
</DocumentFragment>
`;

exports[`app/components/atoms/Checkbox stories Unchecked matches markup snapshot 1`] = `
<DocumentFragment>
  <label
    class="flex items-center font-medium cursor-pointer"
  >
    <input
      class="w-4 h-4 mr-2 cursor-pointer rounded focus:ring-blue-500 focus:ring-1"
      type="checkbox"
    />
     Unchecked Checkbox
  </label>
</DocumentFragment>
`;
